packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -131,9 +148,34 @@ export default class QuanticGeneratedAnswer extends LightningElement {
    * handles disliking the generated answer.
    * @param {CustomEvent} event
    */
-  handleDislike(event) {
+  async handleDislike(event) {
     event.stopPropagation();
     this.generatedAnswer.dislike?.();
+    if (!this.feedbackSubmitted) {
+      // @ts-ignore
+      await FeedbackModal.open({
+        label: this.labels.feedback,
+        size: 'small',
+        description: this.labels.feedback,
+        options: this.options,
+        handleSubmit: this.submitFeedback.bind(this),
+        optionsLabel: this.labels.whyGeneratedAnswerWasNotHelpful,
+      });
+      this.generatedAnswer.closeFeedbackModal();
```

Comment:
We `await` the modal open, to then call the closeFeedbackModal after?

Does that mean all the while the feedback modal is opened, a promise is <pending>?

I might've missed this the first time, I would instead not await anything on open, and also pass a callback function when the modal is closed.


```suggestion
      FeedbackModal.open({
        label: this.labels.feedback,
        size: 'small',
        description: this.labels.feedback,
        options: this.options,
        handleSubmit: this.submitFeedback.bind(this),
        handleClose: () => this.generatedAnswer.closeFeedbackModal(),
        optionsLabel: this.labels.whyGeneratedAnswerWasNotHelpful,
      });
```

Reply:
Unfortunately we don't have the choice but to follow how Salesforce designed the Lightning Modal as documented here: https://developer.salesforce.com/docs/component-library/bundle/lightning-modal/documentation
We can't really pass an onclose callback.